x/poolmanager/types/keys.go

```diff
@@ -67,3 +69,24 @@ func ParseModuleRouteFromBz(bz []byte) (ModuleRoute, error) {
 func KeyPoolVolume(poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%s%d%s", KeyPoolVolumePrefix, KeySeparator, poolId, KeySeparator))
 }
+
+// ParseDenomTradePairKey parses the raw bytes of the DenomTradePairKey into a denom trade pair.
+func ParseDenomTradePairKey(key []byte) (denom0, denom1 string) {
+	keyStr := string(key)
+	parts := strings.Split(keyStr, KeySeparator)
+
+	denom0 = parts[1]
+	denom1 = parts[2]
+
+	err := sdk.ValidateDenom(denom0)
+	if err != nil {
+		panic(err)
```

Comment:
Could we return an error here instead and panic in the caller? I know this is currently only used in genesis but if it were to be switched to state machine / hooks in the future, this might cause fatal problems stemming from panics